File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/Tasks/SleepTask.java
package org.firstinspires.ftc.teamcode.Tasks;

/**
 * Waits for the given amount of time.
 * Unless used in a synchronous task, this will not block the thread of execution.
 * @param ms Time to wait in milliseconds.
 */
public class SleepTask extends Task {
    public long timeAtRun = -1;
    public long ms;

    public SleepTask(long ms) {
        this.ms = ms;
    }

    @Override
    public void tick() {
        long currentTime = System.currentTimeMillis();
        if (currentTime - timeAtRun > ms) state = State.FINISHED;

    }

    @Override
    public void run() {
        timeAtRun = System.currentTimeMillis();
    }
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/tasks/SleepTask.kt
package org.firstinspires.ftc.teamcode.tasks

import org.firstinspires.ftc.teamcode.tasks.Task
import org.firstinspires.ftc.teamcode.tasks.TaskState

class SleepTask(private val ms: Int) : Task() {
    private var timeAtRun: Long = -1
    override fun tick() {
        val currentTime = System.currentTimeMillis()
        if (currentTime - timeAtRun > ms) {
            state = TaskState.FINISHED
            return
        }
    }

    override fun run() {
        timeAtRun = System.currentTimeMillis()
    }
}